### PR TITLE
NAS-107087 / 12.1 / add TCP keepalive settings for remote client

### PR DESCRIPTION
**Pull request by yocalebo**

### CODE REVIEW

File: src/middlewared/middlewared/client/client.py [hunk 1]
```diff
@@ -27,6 +28,25 @@
     LIBZFS = True
 
 
+class TCPKeepAlive(enum.Enum):
+
+    KEEPIDLE = (256, 4)
+    KEEPINTVL = (512, 5)
+    KEEPCNT = (1024, 6)
```
Comment (william-gr) [inline src/middlewared/middlewared/client/client.py:35]: why not use socket.TCP_KEEP*?

Comment (yocalebo) [reply src/middlewared/middlewared/client/client.py:35]: It's not exported in the socket module from what I've gathered.

Comment (william-gr) [reply src/middlewared/middlewared/client/client.py:35]: Maybe not in FreeBSD, but its there for linux at least

Comment (yocalebo) [reply src/middlewared/middlewared/client/client.py:35]: yep, it is in freeBSD too....i must have typo'ed earlier. I'll fix.

**Comment by william-gr** [comment]
@amotin how does that look to you?

**Comment by amotin** [comment]
> @amotin how does that look to you?

It is what I am using in CTL for HA interconnect. Except I am also setting TCP_KEEPINIT to 3.  Works great in my case.

**Pull request merged**